### PR TITLE
96/t/WEB-1997: Fix comparison/numeric filters

### PR DESCRIPTION
**Pull request by ibiziiac-sv**
@codyjca please review. I have ported the changes and the spec, also made some adjustments in the `afterRender` method.

### CODE REVIEW

**Comment by codyjca** [comment]
:+1: thanks Ion!

**Comment by Vicentecarrillo** [comment]
@nburwell do these changes to the datatable filters look safe enough to merge into the release branch before the deploy?
cc: @mikeweaver @deeptidjjain @sherriman

**Comment by nburwell** [comment]
This seems safe to me, especially since the only consumer of the filter code here is the reporting app in web (so, isolated), and it is controlled by a feature flag if issues arise.

I'm assuming it's also been or will be QA'ed before merging too..

File: src/plugins/data_table/filter_view.js [hunk 1]
```diff
@@ -206,6 +211,18 @@ var DataTableFilter = (function(options) {
       this._updateFilterUrlParams();
     },
 
+    _onSelectChange: function (event) {
+      event.stopImmediatePropagation();
+      var filterElementId = event.target.getAttribute('data-filter-id'),
+          previousFilterType = event.target.getAttribute('data-previous-value'),
+          filterType = event.target.value;
```
Comment (nburwell) [inline src/plugins/data_table/filter_view.js]: why not use jQuery for these?

```
var target = $(event.target),
    filterElementId = target.data('filter-id'),
    previousFilterType = target.data('previous-value'),
    filterType = target.val();

...
target.data('previous-value', filterType);
```

**Comment by Vicentecarrillo** [comment]
cool, that was my impression as well. It's being reQA'd at the moment. thanks!

**Pull request closed without merging**